### PR TITLE
fixed meta client cache error

### PR DESCRIPTION
**Pull request by bright-starry-sky**

### CODE REVIEW

**Review by critical27 — APPROVED** [review]
Good job！

File: src/common/clients/meta/MetaClient.cpp [hunk 1]
```diff
@@ -1750,21 +1750,16 @@ MetaClient::getTagSchemaFromCache(GraphSpaceID spaceId, TagID tagID, SchemaVer v
...
-            tagIt->second.empty() ||
-            tagIt->second.size() <= static_cast<size_t>(ver)) {
-            return std::shared_ptr<const NebulaSchemaProvider>();
-        } else if (ver < 0) {
-            return tagIt->second.back();
-        } else {
-            return tagIt->second[ver];
+        if (tagIt != spaceIt->second->tagSchemas_.end() && !tagIt->second.empty()) {
+            if (tagIt->second.size() > static_cast<size_t>(ver)) {
+                return ver < 0 ? tagIt->second.back() : tagIt->second[ver];
```
Comment (panda-sheep) [inline src/common/clients/meta/MetaClient.cpp:1757]: if ver = -1,  What will happen？

Comment (panda-sheep) [reply src/common/clients/meta/MetaClient.cpp:1757]: If ver is -1, tag scehmaVer size() is a normal value ， it will not enter if statement.

Comment (critical27) [reply src/common/clients/meta/MetaClient.cpp:1757]: good catch, :joy: but perf can work with this PR? @bright-starry-sky

**Review by yixinglu — APPROVED** [review]

**Comment by critical27** [comment]
Don't merge, it has bug in it. @bright-starry-sky see the comments.

**Comment by bright-starry-sky** [comment]
Please hold

File: src/common/clients/meta/MetaClient.cpp (hunk 1 above)
Comment (bright-starry-sky) [reply src/common/clients/meta/MetaClient.cpp:1917]: My fault, thanks.

**Comment by bright-starry-sky** [comment]
fixed.

**Review by critical27 — APPROVED** [review]

**Review by panda-sheep — APPROVED** [review]

**Pull request merged**